L1Trigger/L1THGCal/plugins/be_algorithms/HGCalSimCluster.cc

```diff
@@ -0,0 +1,256 @@
+    {
+        private:
+            std::unique_ptr<l1t::HGCalClusterBxCollection> cluster_product;
+            // token
+            edm::EDGetTokenT< std::vector<SimCluster> > sim_token;
+            // handle
+            edm::Handle< std::vector<SimCluster> > sim_handle;
+            // add to cluster
+            void addToCluster(std::unordered_map<uint64_t,std::pair<int,l1t::HGCalCluster> >& cluster_container, uint64_t pid,int pdgid,float  energy,float  eta, float phi)
+            {
+            auto iterator = cluster_container.find (pid);
+            if (iterator == cluster_container.end())
+                {
+                    //create an empty cluster
+                    cluster_container[pid] = std::pair<int,l1t::HGCalCluster>(0,l1t::HGCalCluster());
+                    iterator = cluster_container.find (pid);
+                    iterator -> second . second . setPdgId(pdgid);
+                }
+            // p4 += p4' 
+            math::PtEtaPhiMLorentzVectorD p4;
+            p4.SetPt ( iterator -> second . second . pt()   ) ;
+            p4.SetEta( iterator -> second . second . eta()  ) ;
+            p4.SetPhi( iterator -> second . second . phi()  ) ;
+            p4.SetM  ( iterator -> second . second . mass() ) ;
+            math::PtEtaPhiMLorentzVectorD pp4; 
+            float t = std::exp (- eta);
+            pp4.SetPt ( energy * (1-t*t)/(1+t*t)  ) ;
+            pp4.SetEta( eta ) ;
+            pp4.SetPhi( phi ) ;
+            pp4.SetM  (  0  ) ;
+            p4 += pp4;
+            iterator -> second . second . setP4(p4);
+            return ;
+            }
+
+        protected:
+            using Algorithm<FECODEC>::codec_;
+
+        public:
+            // Constructor
+            //
+            using Algorithm<FECODEC>::Algorithm; 
+            using Algorithm<FECODEC>::name;
+            using Algorithm<FECODEC>::run;
+            using Algorithm<FECODEC>::putInEvent;
+            using Algorithm<FECODEC>::setProduces;
+            using Algorithm<FECODEC>::reset;
+
+            //Consumes tokens
+            HGCalTriggerSimCluster(const edm::ParameterSet& conf,edm::ConsumesCollector&cc) : Algorithm<FECODEC>(conf,cc) { 
+                // I need to consumes the PF Cluster Collection with the sim clustering, TODO: make it configurable (?)
+                // vector<SimCluster>                    "mix"                       "MergedCaloTruth"   "HLT/DIGI"
+                // pf clusters cannot be safely cast to SimCluster
+                sim_token = cc.consumes< std::vector< SimCluster > >(edm::InputTag("mix","MergedCaloTruth","DIGI")); 
+            }
+
+            // setProduces
+            virtual void setProduces(edm::EDProducer& prod) const override final
+            {
+                prod.produces<l1t::HGCalClusterBxCollection>(name());
+            }
+
+            // putInEvent
+            virtual void putInEvent(edm::Event& evt) override final
+            {
+#ifdef DEBUG
```

Comment:
I think I will remove them, but if there is no DEBUG flag defined the compiler will ignore that code.